istio/istio: `BUILD_WITH_CONTAINER=1 make istioctl.completion` fails on MacOS

**Bug description**

Using MacOS,  `BUILD_WITH_CONTAINER=1 make istioctl.completion` fails
```
/work/out/darwin_amd64/istioctl collateral --bash && \
mv istioctl.bash /work/out/darwin_amd64/release/istioctl.bash
/bin/bash: /work/out/darwin_amd64/istioctl: cannot execute binary file: Exec format error
Makefile.core.mk:361: recipe for target '/work/out/darwin_amd64/release/istioctl.bash' failed
```

The issue is that the makefile running on Linux in the container is trying to run a MacOS binary crated during the build process.



**Expected behavior**

`BUILD_WITH_CONTAINER=1 make istioctl.completion` shown run successfully on a Mac

**Steps to reproduce the bug**

Prepare MacOS development as specified.

Run `BUILD_WITH_CONTAINER=1 make istioctl.completion`

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

Current _master_ branch

**How was Istio installed?**

N/A

**Environment where bug was observed (cloud vendor, OS, etc)**

macOS Catalina

--- a/Makefile.core.mk
+++ b/Makefile.core.mk
@@ -86,6 +86,15 @@ export ISTIO_BIN=$(GOBIN)
 
 export ISTIO_OUT:=$(TARGET_OUT)
 export ISTIO_OUT_LINUX:=$(TARGET_OUT_LINUX)
+
+# LOCAL_OUT should include architecture where we are currently running versus the desired.
+# This is used when we need to run a build artifact.
+ifeq ($(IN_BUILD_CONTAINER),1)
+  LOCAL_OUT := $(ISTIO_OUT_LINUX)
+else
+  LOCAL_OUT := $(ISTIO_OUT)
+endif
+
 export HELM=helm
 export ARTIFACTS ?= $(ISTIO_OUT)
 export JUNIT_OUT ?= $(ARTIFACTS)/junit.xml
@@ -197,6 +206,7 @@ default: init build test
 # Downloads envoy, based on the SHA defined in the base pilot Dockerfile
 init: $(ISTIO_OUT)/istio_is_init
 	mkdir -p ${TARGET_OUT}/logs
+	mkdir -p ${TARGET_OUT}/release
 
 # I tried to make this dependent on what I thought was the appropriate
 # lock file, but it caused the rule for that file to get run (which
@@ -365,11 +375,11 @@ ${ISTIO_OUT}/release/istioctl-win.exe: depend
 
 # generate the istioctl completion files
 ${ISTIO_OUT}/release/istioctl.bash: istioctl
-	${ISTIO_OUT}/istioctl collateral --bash && \
+	${LOCAL_OUT}/istioctl collateral --bash && \
 	mv istioctl.bash ${ISTIO_OUT}/release/istioctl.bash
 
 ${ISTIO_OUT}/release/_istioctl: istioctl
-	${ISTIO_OUT}/istioctl collateral --zsh && \
+	${LOCAL_OUT}/istioctl collateral --zsh && \
 	mv _istioctl ${ISTIO_OUT}/release/_istioctl
 
 .PHONY: binaries-test

